spec/src/main/asciidoc/interoperability.asciidoc

```diff
@@ -17,131 +17,136 @@
-and improve the security of a token. The snippet above is just the bare minimum necessary to represent a subject with basic
-information about how tokens should be validated. Also note that the ID Token is not meant to be used as a bearer/access token
-to gain access to resources protected by a resource server or service. Here we propose a bearer/access token format, based on
-some claims from ID Token.
-
-In addition to these standard claims, we also recommend a **preferred_username** claim to carry the username. For instance:
+The JWT can contains all other optional and standard claims:
 ```json
-  {
-      "jti": "5a785440-8cb4-4500-a9fc-0c3aec1e62d5",
-   		"iss": "https://server.example.com",
-   		"sub": "24400320",
-   		"aud": "s6BhdRkqt3",
-   		"nonce": "n-0S6_WzA2Mj",
-   		"exp": 1311281970,
-   		"iat": 1311280970,
-   		"auth_time": 1311280969,
-      "preferred_username": "jdoe",
-	  }
+{
+   "iss": "https://server.example.com",
+   "aud": "s6BhdRkqt3",
+   "exp": 1311281970,
+   "iat": 1311280970,
+   "sub": "24400320",
+   "unique_username <or "upn">": "jdoe@server.example.com",
+   "groups: ["red-group", "green-group", "admin-group"],
+   "roles": ["auditor", "administrator"],
+   "jti": "a-123",
+   "auth_time": 1311280969,
+   "preferred_username": "jdoe",
+   "acr": "phr",
+   "nbf":  1311288970
+        ....... <and all other claims defined by JWT spec and OIDC spec>
 ```
-.Discussion topic
-[NOTE]
-It has been suggested that preferred_username be shortened to something like upn
 
 [NOTE]
-It has also been proposed that we also add some unique user identifier as discussed in JSR-375 to uniquely identify credential.
-If JWT is issued with pairwise "sub" claim, it is not easy to use "sub" as credential, we do need some "unique identifier"
-claim in JWT. How about a new claim "unique_username" or "unique_identifier" or "uid"?
+It has been suggested that a new unique_username/upn be used rather than the OIDC preferred_username due to the fact
+that the OIDC spec states that the Relying Party MUST NOT rely upon this value being unique, as discussed in Section 5.7.
+Can we not just impose the requirement of uniqueness rather than introducing another claim?
 
-From an authorization perspective, we recommend two specific claims to carry role information about a subject.
+An extended form of authorization on a per service basis
 ```json
 	{
    		"iss": "https://server.example.com",
    		"sub": "24400320",
-     "preferred_username": "jdoe",
+       "preferred_username": "jdoe",
    		"aud": "s6BhdRkqt3",
-   		"nonce": "n-0S6_WzA2Mj",
    		"exp": 1311281970,
    		"iat": 1311280970,
    		"auth_time": 1311280969,
-      "realm_access": {
-          "roles": ["role-in-realm", "user", "manager"]
-      },
+      "roles": ["role-in-realm", "user", "manager"],
+      "groups": [],
       "resource_access": {
```

Comment:
For "resource_access", how do you identify "resource server" or "service" for this spec? Do we want "resource server"  or "resource" identifier to be defined in this spec?  If we want "resource server" or "resource" claim in this interop spec,  we do need define this claim. Maybe we could refer to the draft of "Resource Indicators for OAuth 2.0", https://tools.ietf.org/html/draft-campbell-oauth-resource-indicators-02, seems want to standardize "resource" identification, although it is not finalized yet. If it takes longer time for us to define this "resource server" claim, could we focus on the simple JWT format without "resource server" for now, and we come back to address "resource server" in next version spec?